### PR TITLE
Setup `CreationScreen` skeleton

### PR DESCRIPTION
**Pull request by ethangraham2001**
Made some changes to facilitate everyone working on itinerary creation

- Shared `MutableItinerary` class stored in `MapViewModel` with some usage guidelines in the doc
- First usage example in `CreateItineraryMap` that has a button that adds locations in real-time. Not representative of real behavior, but the updates are displaying in real time, and persist after changing screens and back again.
- Small refactor of `HomeNavGraph` and its tests for compatibility
- Fixed `Itinerary.computeCenterOfGravity()`
- Not yet tested, as this is intended purely as a skeleton to be expanded (and tested)

### CODE REVIEW

File: app/src/main/java/com/github/wanderwise_inc/app/model/location/MutableItinerary.kt
Comment (276746) [inline app/src/main/java/com/github/wanderwise_inc/app/model/location/MutableItinerary.kt:1]: why dont we use the builder in the Itinerary data class ? It is almost the same

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+fun CreateItineraryMap(
+    mapViewModel: MapViewModel,
+) {
+  val itinerary = mapViewModel.getNewItinerary()!!
+  val locations = remember { mutableStateListOf<Location>() }
+  for (location in itinerary.locations) {
+    locations += location
+  }
+
+  var ctr = 0
```
Comment (276746) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt:33]: I dont understand what is `ctr`

File: app/src/main/java/com/github/wanderwise_inc/app/model/location/MutableItinerary.kt
Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/model/location/MutableItinerary.kt:1]: Will play around and see if it works

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt (hunk 1 above)
Comment (ethangraham2001) [reply app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt:33]: It's just for the dummy case here. Adds a location and shifts the latitude / longitude for the next point. Just for displaying in-app how the screen works

File: app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt
Comment (YofiY) [inline app/src/main/java/com/github/wanderwise_inc/app/ui/creation/CreateItineraryMap.kt:1]: having locally tried the screen it doesn't really correspond to what the figma displays however the GoogleMap composable is a good start.

**Review by YofiY — APPROVED** [review]

**Pull request merged**